### PR TITLE
Clean `hidden_states_before_norm`

### PR DESCRIPTION
**Pull request by hnyls2002**
This PR unblocks https://github.com/sgl-project/sglang/pull/14046 and clean the `hidden_states_before_norm` logics:

- Remove unnecessary `return_hidden_states_before_norm` flag in schedule batch and logits processor.
- Cleanup pruned hidden states (before and after norm).
- When `return_hidden_states_before_norm` is not needed, do not return it in model forwarding.

### CODE REVIEW

File: python/sglang/srt/managers/schedule_batch.py [hunk 1]
```diff
@@ -2104,7 +2101,6 @@ def get_model_worker_batch(
             dllm_config=self.dllm_config,
             reqs=self.reqs,
             has_grammar=self.has_grammar,
```
[the remark this answers is not in the document]
Comment (hnyls2002) [reply python/sglang/srt/managers/schedule_batch.py:2103]: No, this flag should be derived from worker classes rather than scheduling items (req, batch)
